includes/admin/assets.php

```diff
@@ -180,8 +152,18 @@ function enqueue_editor_assets() {
 			'wp-element',
 			'wp-i18n',
 			'wp-plugins',
-		],
-		AUTOSHARE_FOR_TWITTER_VERSION,
+			'wp-primitives',
+		),
+	);
+	if ( file_exists( $asset_file ) ) {
```

Comment:
@iamdharmesh, is there any reason the asset php file will not be in the final build? I think we should remove this condition.

Reply:
@ravinderk If something goes wrong in the build process and the asset file is not generated then without this check plugin will throw the fatal error. I am inclined to keep this for the safe side, we are following a similar approach in our other plugins as well.

Reply:
@iamdharmesh for this we have e2e tests, right? If we do not want this to happen, then add a test for this, but do not check the file in the production code.

As I said before, there is no code issue but an approach.

cc: @dkotter

Reply:
> @iamdharmesh for this we have e2e tests, right?

@ravinderk Yes, we do have e2e tests. But build files are generated directly in the [deploy action](https://github.com/10up/autoshare-for-twitter/blob/develop/.github/workflows/deploy-to-wpdotorg.yml#L15) and we are not running e2e tests there before deploying the plugin. I am not a fan of adding conditions, but for the safe side, I feel it is worth it here. @dkotter any thoughts on this?

Reply:
@iamdharmesh, we run E2E tests when pushing to `develop` and `trunk`. Both are protective branches. Is this not the same thing?

Reply:
I can see both points here. In theory we shouldn't need a `file_exists` check, as this file should always be there in the final build (and if it's not there, almost certainly we have bigger problems). But as Dharmesh points out, we build the plugin right before deploy, so there's a potential scenario (however slim) that in between building the plugin to run our E2E tests on a push to `trunk` and running the same build process when tagging a release, that something may go wrong (for instance, maybe a package we use in our build process has a new release that breaks something).

Again, this is all very unlikely that our E2E tests will pass on a merge to `trunk` but then something goes wrong during the release tag process but it could happen. Because of that, I don't mind having this check as a means of defensive programming (though ideally we're also checking the final release files that make it to .org to ensure all built files exist there)